app/models/concerns/taggable.rb

```diff
@@ -77,6 +82,32 @@ def freeform_string=(tag_string)
     parse_tags(Freeform, tag_string)
   end
 
+  # Process a string of tags from any tag class. Unlike #parse_tags, it allows the creation of UnsortedTag instances.
+  # Used in bookmarks and collections.
+  def tag_string=(tag_string)
+    # Make sure that we trigger the callback for our taggings.
+    self.taggings.destroy_all
+
+    self.tags = []
+
+    # Replace unicode full-width commas
+    tag_string.gsub!(/\uff0c|\u3001/, ",")
+    tag_array = tag_string.split(ArchiveConfig.DELIMITER_FOR_INPUT)
+
+    tag_array.each do |string|
+      string.strip!
+      next if string.blank?
+      
+      if (tag = Tag.find_by_name(string))
```

Comment:
I think the [Style/ConditionalAssignment](https://docs.rubocop.org/rubocop/cops_style.html#styleconditionalassignment) warning means we should do something like:
```rb
self.tags << if (tag = Tag.find_by_name(string))
```

Reply:
What would you do with the else statement, if the "if" is in line like that? I think that warning is for a different scenario, where you're assigning a variable to a conditional, to use later in the code. In this case, we need the variable inside the conditional, so assigning it from the entire statement isn't appropriate.

Since it's a pretty simple condition, it might be cleanest to convert it to a ternary, something like this: 
```
tag = Tag.find_by_name(string)
tag.present? ? self.tags << tag : self.tags << UnsortedTag.create(name: string)
```
Does that sound alright?

Reply:
If you run `rubocop -a` on the code that you have now, it becomes:
```
      self.tags << if (tag = Tag.find_by_name(string))
                     tag
                   else
                     UnsortedTag.create(name: string)
                   end
```
Which I think is what red is suggesting (and Hound is suggesting).

If instead you run `rubocop -a` on the original code (with `tag = Tag.find_by_name(string)` as a separate line, instead of as the condition of the `if`), it becomes:
```
      tag = Tag.find_by_name(string)
      self.tags << (tag || UnsortedTag.create(name: string))
```
Though of course, the absolute simplest way to rewrite it is:
```
self.tags << (Tag.find_by_name(string) || UnsortedTag.create(name: string))
```

Reply:
Awesome. Thanks for the explanation. :)